src/sections/mod.rs

```diff
@@ -119,12 +119,17 @@ impl<'a> PsdCursor<'a> {
     }
 
     /// Get the cursor's position
-    pub fn position(&self) -> u64 {
-        self.cursor.position()
+    pub fn position(&self) -> usize {
+        // since we are a cursor over a [u8]
+        // that means the total amount of bytes is less than isize::MAX
+        // therefore the biggest index is isize::MAX
+        // also meaning that it can be safely cast to a usize
+        self.cursor.position() as usize
     }
 
-    pub fn seek(&mut self, pos: u64) {
-        self.cursor.set_position(pos);
+    pub fn seek(&mut self, pos: usize) {
+        // see Self::position for more info
```

Comment:
More info about what? Unless you're on a 128 bit architecture and your PSD has more than `u64::MAX` bytes this cast is always ok.

Reply:
Well yeah I'm explaining why its fine, I'm saying if its in memory that for now (IIRC rust doesnt support 128 bit arch yet) means `pos < u64::MAX` so the cast is fine